File: Easy/566.重塑矩阵.cpp
/*
 * @Author: Boredcui
 * @Date: 2022-03-03 14:46:59
 * @LastEditors: Boredcui
 * @LastEditTime: 2022-03-03 14:54:22
 * @FilePath: \LeetCode\566.重塑矩阵.cpp
 * @Description:
 *
 * Copyright (c) 2022 by boredcui, All Rights Reserved.
 */
/*
 * @lc app=leetcode.cn id=566 lang=cpp
 *
 * [566] 重塑矩阵
 */

// @lc code=start
class Solution
{
public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &mat, int r, int c)
    {
        int m = mat.size();
        int n = mat[0].size();

        if (m * n != r * c)
        {
            return mat;
        }

        vector<vector<int>> ans(r, vector<int>(c));
        for (int x = 0; x < m * n; ++x)
        {
            ans[x / c][x % c] = mat[x / n][x % n];
        }
        return ans;
    }
};
// @lc code=end

File: Easy/566.重塑矩阵.py
'''
Author: Boredcui
Date: 2022-03-03 14:13:40
LastEditors: Boredcui
LastEditTime: 2022-03-03 14:45:46
FilePath: \LeetCode\566.重塑矩阵.py
Description: 

Copyright (c) 2022 by boredcui, All Rights Reserved. 
'''
#
# @lc app=leetcode.cn id=566 lang=python3
#
# [566] 重塑矩阵
#

# @lc code=start


class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m = len(mat)
        n = len(mat[0])
        if m*n != r*c:
            return mat
        nat = [[0] * c for _ in range(r)]
        for x in range(m*n):
            nat[x//c][x % c] = mat[x//n][x % n]
        return nat
# @lc code=end
